Test-Source/Invoke-ExecCommandCore.Tests.ps1

```diff
@@ -0,0 +1,98 @@
+Import-Module -Force $PSScriptRoot/../Source/Docker-CI.psm1
+Import-Module -Force $PSScriptRoot/Docker-CI.Tests.psm1
+
+. "$PSScriptRoot\..\Source\Private\Invoke-Command.ps1"
+. "$PSScriptRoot\..\Source\Private\CommandCoreResult.ps1"
+. "$PSScriptRoot\..\Source\Private\New-Process.ps1"
+
+Describe 'Runs only external tools' {
+
+    Context 'Runs a simple external command' {
+        if ($IsWindows) {
+            $command = [PSCustomObject]@{
+                'Command'     = 'find'
+                'CommandArgs' = '/?'
+            }
+        } elseif ($IsLinux) {
+            $command = [PSCustomObject]@{
+                'Command'     = 'grep'
+                'CommandArgs' = '--help'
+            }
+        }
+
+        BeforeEach {
+            $tempFile = New-TemporaryFile
+        }
+
+        AfterEach {
+            Remove-Item $tempFile -Force
+        }
+
+        It 'returns correct output and exit code, silently' {
+            $result = Invoke-Command -Command $command.Command -CommandArgs $command.CommandArgs -Quiet:$true 6> $tempFile
+            $result.ExitCode | Should -Be 0
+            $result.Output | Should -Not -BeNullOrEmpty
+            Get-Content $tempFile | Should -BeNullOrEmpty
+        }
+
+        It 'can run a command with no args' {
+            $result = Invoke-Command -Command $command.Command -Quiet:$true
+            $result.ExitCode | Should -Not -Be 0
+        }
+
+        It 'returns correct output and exit code, verbosely' {
+            $result = Invoke-Command -Command $command.Command -CommandArgs $command.CommandArgs -Quiet:$false 6> $tempFile
+            $result.ExitCode | Should -Be 0
+            $result.Output | Should -Not -BeNullOrEmpty
+            Get-Content $tempFile | Should -Not -BeNullOrEmpty
+        }
+
+        It 'returns the exit code for failing commands, silently' {
+            $result = Invoke-Command -Command $command.Command -CommandArgs "---nope-this-is-clearly-wrong" -Quiet:$true 6> $tempFile
+            $result.ExitCode | Should -Not -Be 0
+            $result.Output | Should -Not -BeNullOrEmpty
+            Get-Content $tempFile | Should -BeNullOrEmpty
+        }
+
+        It 'returns the exit code for failing commands, verbosely' {
+            $result = Invoke-Command -Command $command.Command -CommandArgs "---nope-this-is-clearly-wrong" -Quiet:$false 6> $tempFile
+            $result.ExitCode | Should -Not -Be 0
+            $result.Output | Should -Not -BeNullOrEmpty
+            Get-Content $tempFile | Should -Not -BeNullOrEmpty
+        }
+    }
+
+    Context 'Run a mocked command' {
+
```

Comment:
Looks like round 2 of the PR. Some comments to remind ourselves about it?
```suggestion
	# TODO: Add more test
```

Reply:
It's a reminder for a follow-up PR I think should be, it's described better in #94